src/runtime.js

```diff
@@ -244,9 +245,14 @@ function adoptShared(global, opts, callback) {
    */
   function installExtension(fnOrStruct) {
     if (typeof fnOrStruct == 'function') {
-      fnOrStruct(global.AMP);
+      const fn = fnOrStruct;
+      chunk(global.document, () => fn(global.AMP));
```

Comment:
Important! This will immediately break shadow multi-doc mode since this is run before an ampdoc is available.